fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java

```diff
@@ -992,12 +992,14 @@ protected void scanAndLoad() {
 
             if (!this.cosBucketList.isEmpty()) {
                 // Process using COS as our source data repository
-                if (cosProperties != null && cosProperties.size() > 0) {
-                    cosClient = new COSClient(cosProperties);
-                } else {
-                    throw new IllegalArgumentException("COS configuration required");
+                if (cosClient == null) {
```

Comment:
I wonder if we should have the COSClient initialization just in one place. The issue is how we decide whether or not we need it. Perhaps a method that creates it on demand if it doesn't exist?

Reply:
my we just add it to the current `configure()` method an instantiate it whenever we have some `cosProperties` passed to us?

in main, configure() is called right before process(), and that is where we set up the db stuff already...

Reply:
ok, i pushed a commit that does this...let me know what you think